src/gui/glRenderer.js

```diff
@@ -278,6 +278,13 @@ createNameSpace("realityEditor.gui.glRenderer");
         canvas.style.height = canvas.height + 'px';
         gl = canvas.getContext('webgl2');
 
+        realityEditor.device.layout.onWindowResized(({width, height}) => {
+            canvas.style.width = width + 'px';
+            canvas.style.height = height + 'px';
+            // you would think you should update canvas.width and canvas.height too, but that breaks
+            // the coordinate system, for a reason I don't understand. Just update style.width and style.height
```

Comment:
This mostly makes sense given that there's no mechanism for sending the new canvas pixel dimensions (i.e. canvas.width and canvas.height) to the proxied frame. If there were, then the proxied frame could call threejs's setSize function or otherwise update their own viewport to match up. The approach you're taking is a good compromise since most people won't be doing things like resizing from super thin to super wide

Reply:
Thanks for the explanation!